design/baremetal-operator/bulk-set-bios-config.md

```diff
@@ -0,0 +1,414 @@
+- If there is no reference, the schema data for all BIOS settings for the
+  host will be stored locally
+- A hash will be generated from this stored data, if a corresponding hash
+  does not exist in the `FirmwareSchema` CRs, a new CR will be created and
+  the schema data will be copied to it
+- The reference to the `FirmwareSchema` resource in `HostFirmwareSettings`
+  will be updated with the hash. A suggested reference string is to
+  include the vendor and model in the string, such as
+  "vendor,model,64 chars of hash"
+- Using the hash will ensure that the schemas match. In the case of
+  hardware from the same vendor and model, but with a different firmware
+  version, it is possible that the schemas would differ. In this case a new
+  `FirmwareSchema` will be created.
+- The name of the `FirmwareSchema` resource will be added to the Labels
+  for the `HostFirmwareSettings` so that it can be used to delete the
+  `FirmwareSchema` resource when it is no longer being used.
+
+An example of the proposed CRD for `HostFirmwareSettings` is as follows:
+
+```yaml
+---
+kind: CustomResourceDefinition
+metadata:
+  name: hostfirmwaresettings.metal3.io
+spec:
+  group: metal3.io
+  names:
+    kind: HostFirmwareSettings
+    listKind: HostFirmwareSettingsList
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+          spec:
+            description: HostFirmwareSettingsSpec defines the desired state
+            properties:
+              vendorSettings:
+                items:
+                  description: describes a setting whose value can be changed
+                  properties:
+                    name:
+                      description: Identifier for the Firmware setting.
+                      type: string
+                    value:
+                      description: Value of the Firmware setting.
+                      type: string
+                  required:
+                  - name
+                  - value
+                  type: object
+                type: array
+            type: object
+          status:
+            description: defines the observed state of HostFirmwareSettings
+            properties:
+              settingsData:
+                items:
+                  description: describes current settings and their schema
+                  properties:
+                    name:
+                      description: Identifier for the Firmware setting.
+                      type: string
+                    value:
+                      description: Value of the Firmware setting.
+                      type: string
+                    schemaReference:
+                      description: Reference to resource containing schema
+                      type: string
+                type: array
+            type: object
+```
+
+An example of the proposed CRD for `FirmwareSchema` is as follows:
+
+```yaml
+---
+kind: CustomResourceDefinition
+metadata:
+  name: hostfirmwaresettings.metal3.io
```

Comment:
I guess it's typo here (and below): hostfirmwareschema?

Reply:
Yes. Fixed the typo